core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java

```diff
@@ -699,7 +701,16 @@ else if (Tables.getTableState(context.getInstance(), table) == TableState.OFFLIN
 
     }
 
-    void addMutations(MutationSet mutationsToSend) {
+    void queueMutations(final MutationSet mutationsToSend) {
+      queueThreadPool.submit(new Runnable() {
```

Comment:
> Are you suggesting re-using the existing ExecutorService?

Nope. I'm saying just make a single thread which is reading off some queue. The clients would be submitting mutations onto that queue. You just have one thread always running, looking for mutations to add. You avoid the whole creating, scheduling and starting the Runnable.